### PR TITLE
APP-474: Review fixes

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @GamovCoder

### CODE REVIEW

**Comment by Slash32** [comment]
@kaa-leeroy test this please

File: notification/source/c/src/kaa_demo.c [hunk 1]
```diff
@@ -65,35 +66,26 @@ static int read_topic_id(void)
     return atoi(input);
```
Comment (ohamov) [inline notification/source/c/src/kaa_demo.c]: Here you still have a problem. `atoi()` returns `int`.

File: notification/source/c/src/kaa_demo.c [hunk 2]
```diff
@@ -65,35 +66,26 @@ static int read_topic_id(void)
...
+    return t->id == *(int64_t*)context;
 }
 
 static void on_notification(void *context, uint64_t *topic_id, kaa_notification_t *notification)
 {
     (void)context;
     if (notification->alert_message) {
         kaa_string_t *message = (kaa_string_t *)notification->alert_message;
-        demo_printf("Notification for topic id '%lu' received\r\n", *topic_id);
+        demo_printf("Notification for topic id '%li' received\r\n", *topic_id);
```
Comment (ohamov) [inline notification/source/c/src/kaa_demo.c]: `%li` is for `long int`. `sizeof(long int)` for x86_64 is 4. Have you seen my previous comment?

File: notification/source/c/src/kaa_demo.c [hunk 3]
```diff
@@ -111,7 +103,7 @@ static void show_topics(kaa_list_t *topics)
     kaa_list_node_t *it = kaa_list_begin(topics);
     while (it) {
         kaa_topic_t *topic = (kaa_topic_t *)kaa_list_get_data(it);
-        demo_printf("Topic: id '%lu', name: %s, type: ", topic->id, topic->name);
+        demo_printf("Topic: id '%li', name: %s, type: ", topic->id, topic->name);
```
Comment (ohamov) [inline notification/source/c/src/kaa_demo.c]: same

File: notification/source/c/src/kaa_demo.c [hunk 4]
```diff
@@ -140,7 +132,7 @@ void on_topics_received(void *context, kaa_list_t *topics)
             if (err) {
                 demo_printf("Failed to subscribe.\r\n");
             } else {
-                demo_printf("Subscribed to optional topic '%lu'\r\n", topic->id);
+                demo_printf("Subscribed to optional topic '%li'\r\n", topic->id);
```
Comment (ohamov) [inline notification/source/c/src/kaa_demo.c]: same

**Review by ohamov — CHANGES_REQUESTED** [review]
There are comments which haven't been addressed.

File: notification/source/c/src/kaa_demo.c [hunk 5]
```diff
@@ -46,7 +47,7 @@ static const char *color_to_str(enum color code_color)
 
 #define INPUT_LENGTH 8
 
-static int read_topic_id(void)
+static int64_t read_topic_id(void)
```
Comment (ohamov) [inline notification/source/c/src/kaa_demo.c]: topic id type is not `int64_t`

File: notification/source/c/src/kaa_demo.c [hunk 6]
```diff
@@ -65,35 +66,26 @@ static int read_topic_id(void)
...
-    while (it) {
-        kaa_topic_t *t = (kaa_topic_t *)kaa_list_get_data(it);
-        if (t->id == topic_id) {
-            return t;
-        }
-        it = kaa_list_next(it);
-    }
-    return NULL;
+    kaa_topic_t *t = (kaa_topic_t *)data;
+    return t->id == *(int64_t*)context;
```
Comment (GamovCoder) [inline notification/source/c/src/kaa_demo.c]: topic id type is `unint64_t`

File: notification/source/c/src/kaa_demo.c (hunk 5 above)
Comment (RostakaGmfun) [reply notification/source/c/src/kaa_demo.c]: But in #481  you've said, it is `int64_t`. Also, `int64_t` is used for topic ids in C++ SDK. (correct me if I'm wrong)

Comment (ohamov) [reply notification/source/c/src/kaa_demo.c]: Yep, I looked at C++ SDK. Anyway, in C it is `uint64_t` type. so fix it please

File: notification/source/c/src/kaa_demo.c (hunk 1 above)
Comment (RostakaGmfun) [reply notification/source/c/src/kaa_demo.c]: I will rebase on master as soon as #484 is merged

**Review by ohamov — APPROVED** [review]
LGTM

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Review by Slash32 — APPROVED** [review]

**Pull request merged**